File: TP01/Arquivo.java
/*
AEDS 2 TP01
Daniel Felipe Coelho de Freitas
matricula 859230
*/
import java.io.RandomAccessFile;

public class Arquivo {
    static void escreveArquivo(int n, String nomeArquivo) throws Exception{
        RandomAccessFile arquivo = new RandomAccessFile("saida.txt", "rw");
        for(int i = 0; i < n; i++){
            arquivo.writeDouble(MyIO.readDouble());
        }
        arquivo.close();
       
    }
    static void leArquivo(int n, String nomeArquivo) throws Exception{
        RandomAccessFile arquivo = new RandomAccessFile("saida.txt", "r");
        double valor;
        for(int i = 8*(n-1); i >= 0; i-=8){
            arquivo.seek(i);
            valor = arquivo.readDouble();
            if(valor == (int)valor){
                MyIO.println((int)valor);
            }else{
                MyIO.println(valor);
            }
        }
        arquivo.close();
    }
    public static void main(String[] args) {
        String saida = "saida.txt";
        int numero = MyIO.readInt();
        try{
            escreveArquivo(numero, saida);
            leArquivo(numero, saida);
        }catch(Exception e){

        }
        
    }
}

File: TP01/arquivo.c
/*
AEDS 2 TP01
Daniel Felipe Coelho de Freitas
matricula 859230
*/
#include <stdio.h>
void escreveArquivo(int n, char *nomeArquivo){
        FILE *arquivo;
        double numero = 0;
        arquivo = fopen(nomeArquivo, "w");//cria o arquivo para escrita, se ja existir um arquivo com o nome ele limpa o arquivo
        for(int i = 0; i < n; i++){
            scanf("%lf", &numero);//le o numero para escrever no arquivo
            fprintf(arquivo, "%g\n",numero);//escreve o numero no arquivo
        }
        fclose(arquivo);//fecha o arquivo
       
    }
void leArquivo(FILE *arquivo){
        double valor;
        fscanf(arquivo, "%lf", &valor);//le o valor
        if(!feof(arquivo)){
            leArquivo(arquivo);//chama recursivamente a funcao ate chegar o fim do arquivo e fazer o print do ultimo ao primeiro
            printf("%g\n",valor);//%g faz o print do valor em inteiro ou decimal sem os zeros a mais
        }
        
    }
int main(int argc, char const *argv[]){
    int num;
    char saida[10] = "saida.txt";
    scanf("%d", &num);//le a quantidade de numeros a ser escritos no arquivo
    escreveArquivo(num, saida);
    FILE *arquivo;
    arquivo = fopen(saida, "r");//abre o arquivo para leitura
    leArquivo(arquivo);
    fclose(arquivo);//fecha o arquivo
    return 0;
}
